lib/go/thrift/simple_server_test.go

```diff
@@ -154,3 +158,112 @@ func TestNoHangDuringStopFromDanglingLockAcquireDuringAcceptLoop(t *testing.T) {
 	runtime.Gosched()
 	serv.Stop()
 }
+
+func TestNoHangDuringStopFromClientNoDataSendDuringAcceptLoop(t *testing.T) {
+	ln, _ := net.Listen("tcp", "localhost:0")
+
+	proc := &mockProcessor{
+		ProcessFunc: func(in, out TProtocol) (bool, TException) {
+			in.ReadMessageBegin(context.Background())
+			return false, nil
+		},
+	}
+
+	trans := &mockServerTransport{
+		ListenFunc: func() error {
+			return nil
+		},
+		AcceptFunc: func() (TTransport, error) {
+			conn, err := ln.Accept()
+			if err != nil {
+				return nil, err
+			}
+
+			return NewTSocketFromConnConf(conn, nil), nil
+		},
+		CloseFunc: func() error {
+			return nil
+		},
+		InterruptFunc: func() error {
+			return ln.Close()
+		},
+	}
+
+	serv := NewTSimpleServer2(proc, trans)
+	go serv.Serve()
+	time.Sleep(10 * time.Millisecond)
+
+	netConn, err := net.Dial("tcp", ln.Addr().String())
+	if err != nil || netConn == nil {
+		t.Fatal("error when dial server")
+	}
+	time.Sleep(10 * time.Millisecond)
+
+	st := time.Now()
+	ServerCloseTimeout = 50 * time.Millisecond
```

Comment:
nit: when a test involves changing the value of a global variable, I'd prefer to change its value back so we don't have tests relying on the default value becomes flaky because of the order of test executions, so this should be something like:
```suggestion
	backupServerCloseTimeout = ServerCloseTimeout
	t.Cleanup(func() {
		ServerCloseTimeout = backupServerCloseTimeout
	})
	ServerCloseTimeout = 50 * time.Millisecond
```

same for the other test.